src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Interop/MethodOptions.ts

```diff
@@ -0,0 +1,28 @@
+﻿export interface MethodOptions {
+    type: TypeInstance;
+    method: MethodInstance;
+    async?: { resolveId: string, rejectId: string, functionName: string }
+}
+
+export interface MethodInstance {
+    name: string;
+    typeArguments: { [key: string]: TypeInstance }
+    parameterTypes: TypeInstance[];
+}
+
+export interface TypeInstance {
+    assembly: string;
+    typeName: string;
+    typeArguments: { [key: string]: TypeInstance };
+}
+
+export interface DotnetMethodArgumentsList {
```

Comment:
TBH it doesn't look like the interface is involved in any type checking anywhere. Maybe it isn't required.